src/script/src/python/pyo3/dataframe_impl.rs

```diff
@@ -251,25 +262,26 @@ impl PyExpr {
             CompareOp::Gt => DfExpr::gt,
             CompareOp::Ge => DfExpr::gt_eq,
         };
-        Ok(op(self.inner.clone(), other.inner.clone()).into())
+        py.allow_threads(|| Ok(op(self.inner.clone(), other.inner.clone()).into()))
     }
     fn alias(&self, name: String) -> PyResult<PyExpr> {
         Ok(self.inner.clone().alias(name).into())
     }
     fn __and__(&self, py: Python<'_>, other: PyExprRef) -> PyResult<PyExpr> {
-        Ok(self
-            .inner
-            .clone()
-            .and(other.borrow(py).inner.clone())
-            .into())
+        let other = other.borrow(py).inner.clone();
+        py.allow_threads(|| Ok(self.inner.clone().and(other).into()))
```

Comment:
Are there any best practices for when to use or not to use `py.allow_threads`?

Reply:
> Are there any best practices for when to use or not to use `py.allow_threads`?

The doument says this enable [Parallelism](https://pyo3.rs/v0.13.2/parallelism?search=allow_threads#parallelism)
This method releases the Global Interpreter Lock (GIL) in Python, allowing other Python threads to run while the Rust thread continues executing. This is useful if you have a long-running Rust computation that would otherwise block the Python interpreter and prevent other threads from running.